cpp/example_code/general/CMakeLists-minimal.txt

```diff
@@ -1,18 +1,26 @@
-# minimal CMakeLists.txt for the AWS SDK for C++
+# Minimal CMakeLists.txt for the AWS SDK for C++.
 cmake_minimum_required(VERSION 3.2)
 
+# Build using the C++ standard version 11.
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+# Use shared libraries, which is the default for the AWS C++ SDK build.
+option(BUILD_SHARED_LIBS "Build shared libraries" ON)
+
 # "my-example" is just an example value.
 project(my-example)
 
+# By default the AWS SDK for C++ is installed into /usr/local.
+# You may need to adjust the default include path on some platforms.
+# include_directories(/usr/local/include)

```

Comment:
The `find_package` command is responsible for determining where the AWS sdk is installed.
When using `target_link_libraries` with `AWSSDK_LINK_LIBRARIES`, include directories should automatically be set up for your target.

Reply:
I swear this wasn't working yesterday, but removing this now works. I am confused to as why, but what you say makes sense, I'm removing this.

Reply:
Maybe you were not linking with `AWSSDK_LINK_LIBRARIES`, or had a different list of components in `find_package`?

From what I saw in the install cmakelists it should work properly.
There is however something that could have happened: if you didn't put find_package yet and used the name of the library or component then it could have found it if it was somehow in your libraries path.
It's linked to https://github.com/aws/aws-sdk-cpp/issues/1307 where if you do not specify a namespace, cmake can fallback to just giving the name of the library to the linker instead of using a target. Then it would link, but properties would not propagate since it's not really linking a target.